src/js/application.js

```diff
@@ -0,0 +1,288 @@
+            this.setLanguage(savedLanguage, false);
+        }
+        this.loadLocationByIP();
+    }
+
+    showErrorMessage() {
+        const ERROR_MESSAGE_ELEMENT = document.querySelector(selectors.ERROR_MESSAGE);
+        ERROR_MESSAGE_ELEMENT.style.display = 'block';
+    
+        setTimeout(() => {
+            ERROR_MESSAGE_ELEMENT.style.display = 'none';
+        }, 3000);    
+    }
+
+    async onLocationChange(e) {    
+        e.preventDefault();
+
+        const SEARCH_INPUT_ELEMENT = document.querySelector(selectors.SEARCH_INPUT_FIELD);
+        const location = await Location.getLocationByName(SEARCH_INPUT_ELEMENT.value, this.language);
+
+        if (location) {
+            await location.loadLocationTimezone();
+            await this.setLocation(location);
+        } else {
+            this.showErrorMessage();
+        }
+    }
+
+    showDate() {
+        const DATE_ELEMENT = document.querySelector(selectors.DATE);
+
+        DATE_ELEMENT.innerHTML = this.currentLocation.getDate();
+    }
+
+    showTime() {
+        const TIME_ELEMENT = document.querySelector(selectors.TIME);
+
+        TIME_ELEMENT.innerHTML = this.currentLocation.getTime();
+        setTimeout(this.showTime.bind(this), 1000);
+    }
+
+    async loadLocationByIP() {
+        let response = await fetch('https://ipinfo.io/json?token=edaa4c28dad526');
+
+        response = await response.json();
+        const location = await Location.getLocationByName(response.city, this.language);
+        if (location) {
+            await this.setLocation(location);
+        }
+    }
+
+    async setLocation(location) {
+        const LATITUDE_DEGREES_ELEMENT = document.querySelector(selectors.LATITUDE_DEGREES);
+        const LONGITUDE_DEGREES_ELEMENT = document.querySelector(selectors.LONGITUDE_DEGREES);
+        const LATITUDE_MINUTES_ELEMENT = document.querySelector(selectors.LATITUDE_MINUTES);
+        const LONGITUDE_MINUTES_ELEMENT = document.querySelector(selectors.LONGITUDE_MINUTES);
+        const LOCATION_ELEMENT = document.querySelector(selectors.LOCATION);
+        const MINUTES_PER_DEGREES = 60;
+
+        this.currentWeather = new Weather(location);
+        await this.currentWeather.loadWeather();
+        this.currentLocation = location;        
+    
+        LATITUDE_DEGREES_ELEMENT.innerHTML = Math.round(location.latitude);
+        LONGITUDE_DEGREES_ELEMENT.innerHTML = Math.round(location.longitude);
+        LATITUDE_MINUTES_ELEMENT.innerHTML = Math.round((location.latitude - Math.trunc(location.latitude)) * MINUTES_PER_DEGREES);
+        LONGITUDE_MINUTES_ELEMENT.innerHTML = Math.round((location.longitude - Math.trunc(location.longitude)) * MINUTES_PER_DEGREES);
+        LOCATION_ELEMENT.innerHTML = `${this.currentLocation.cityName}, ${this.currentLocation.country}`;
+
+        this.showMap();
+        this.showDate();
+        this.showTime();
+        this.showLoadedWeather(this.currentWeather);
```

Comment:
why do you pass some value from this as parameter?

Reply:
fixed